### PR TITLE
Use replace() instead of add() function to add fragments to backstack

### PR DESCRIPTION
**Pull request by ancirja-m**

### CODE REVIEW

**Review by CesarValiente — APPROVED** [review]
LGMT! Thanks Andrei!

**Review by soalb-m — APPROVED** [review]

File: navigation/navigation-fragment/src/main/java/androidx/navigation/fragment/FoldableFragmentNavigator.java [hunk 1]
```diff
@@ -214,6 +217,7 @@ public FoldableNavDestination navigate(@NonNull Destination destination, @Nullab
         // The commit succeeded, update our view of the world
         if (isAdded) {
             mBackStack.add(destId);
+            Log.d("ANDREI", "BackStackSize = " + mBackStack.size());
```
Comment (bimiron) [inline navigation/navigation-fragment/src/main/java/androidx/navigation/fragment/FoldableFragmentNavigator.java:220]: Do we need to keep this log or can it be removed?

**Review by bimiron — APPROVED** [review]
Looks good!

File: navigation/navigation-fragment/src/main/java/androidx/navigation/fragment/FoldableFragmentNavigator.java (hunk 1 above)
Comment (CesarValiente) [reply navigation/navigation-fragment/src/main/java/androidx/navigation/fragment/FoldableFragmentNavigator.java:220]: wops! nice catch!

Comment (ancirja-m) [reply navigation/navigation-fragment/src/main/java/androidx/navigation/fragment/FoldableFragmentNavigator.java]: No. My mistake. 😆

**Review by CesarValiente — APPROVED** [review]
LGTM! Thanks Andrei!

**Pull request merged**